RapidReact/src/main/java/frc/robot/subsystem/ClimberSubsystem.java

```diff
@@ -329,6 +332,38 @@ public void periodic() {
     climberRightPosition.log(LogLevel.GENERAL, climberRight.getSelectedSensorPosition());
     climberRightError.log(LogLevel.GENERAL, climberRight.getClosedLoopError());
 
+    boolean climbLeftRevLimitSwitchClosed = climberLeft.getSensorCollection().isRevLimitSwitchClosed();
+    boolean climbRightRevLimitSwitchClosed = climberRight.getSensorCollection().isRevLimitSwitchClosed();
+
+    if (!climbLeftRevLimitSwitchClosed)
+    {
+      if (!climbLeftEncoderZeroed)
+      {
+        climberLeft.set(ControlMode.PercentOutput, climbRetractSlow.currentValue());
+      }
+    }
+    else
+    {
+      climberLeft.setSelectedSensorPosition(0);
+      climberLeft.set(0);
```

Comment:
As far as I can tell, this sets the climber speed to zero whenever the limit switch is closed, which I don't think is your desired behavior. You can draw out a finite state machine state diagram for just the zeroing process, if you want to avoid logic bugs like this.

Reply:
wait then what else would we want to do?

Reply:
How can you ever extend the climber if the motor speed is set to zero whenever the retracted limit switch is pressed.

The correct behavior is setting the speed to zero once, at the same time as the isZeroed variable is set to false, then relying on the motor controller's configured limit switch behavior (not allowing + output when + switch is pressed and vise versa for - switch).

Reply:
> at the same time as the isZeroed variable is set to false

this is just at init, did you mean when it's set to true?

also won't the limit switch make the motor stop anyways?